dwitter/views_v2.py

```diff
@@ -17,10 +20,17 @@
 from dwitter.utils import length_of_code
 
 
-class UserViewSet(mixins.RetrieveModelMixin, viewsets.GenericViewSet):
+class UserViewSet(viewsets.ModelViewSet):
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
+    def get_permissions(self):
+        # POST action to create new user need to be available to everyone
+        if self.request.method == 'POST':
```

Comment:
Does this also set AllowAny for detail actions like `set_email`? Perhaps we also need to check for `self.action == 'create'` as well here?

Reply:
Good call; added some tests, and this was not an issue in practice since both `set_email` and `set_password` explicitly checks who is authorized. But adding the action check is still the right thing to do. Added.